### PR TITLE
Router

### PR DESCRIPTION
**Pull request by d33d33**

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -544,6 +546,16 @@ fn load_path<P: AsRef<Path>>(file_path: P, config: &mut Config) -> Result<(), Co
                     try!(cast::u64(timeout).map_err(|_| format!("parameters.timeout is invalid")));
                 config.parameters.timeout = timeout;
             }
+            if !doc["parameters"]["router-parallel"].is_badvalue() {
+                let router_parallel = try!(
```
Comment (FlorentinDUBOIS) [inline src/config.rs]: use ? instead of try!

File: src/config.rs [hunk 2]
```diff
@@ -544,6 +546,16 @@ fn load_path<P: AsRef<Path>>(file_path: P, config: &mut Config) -> Result<(), Co
...
+                    doc["parameters"]["router-parallel"]
+                        .as_i64()
+                        .ok_or("parameters.router-parallel should be a number")
+                );
+                let router_parallel =
+                    try!(cast::u64(router_parallel).map_err(|_| format!("parameters.router-parallel is invalid")));
```
Comment (FlorentinDUBOIS) [inline src/config.rs]: use ? instead of try!

File: src/main.rs [hunk 3]
```diff
@@ -133,19 +133,8 @@ fn main() {
...
-            config.parameters.clone(),
-            sigint.clone(),
-        );
-        handles.push(thread::spawn(move || {
-            slog_scope::scope(&slog_scope::logger().new(o!()), || {
-                router::router(&sinks, &labels, &parameters, sigint)
-            });
-        }));
-    }
+    let mut router = router::Router::new(&config.sinks, &config.parameters, &config.labels); // FIXME
```
Comment (FlorentinDUBOIS) [inline src/main.rs]: Why FIXME ?

**Comment by ovh-cds** [comment]
CDS Report build-rust-multi#92.0 ✘  
* Cargo
  * CentOS:6 ✘  
  * CentOS:7 ✘  
  * Debian:Jessie -  
  * Debian Stretch -  
  * Debian:Wheezy -  
  * Docker package ✘  
  * Ubuntu:Bionic ✘  
  * Ubuntu:Trusty -  
  * Ubuntu:Vivid -  
  * Ubuntu:Xenial -  
  * Ubuntu:Yakkety -

**Comment by ovh-cds** [comment]
CDS Report build-rust-multi#94.0 ✘  
* Cargo
  * CentOS:6 ✘  
  * CentOS:7 -  
  * Debian:Jessie -  
  * Debian Stretch -  
  * Debian:Wheezy -  
  * Docker package ✔ 
  * Ubuntu:Bionic -  
  * Ubuntu:Trusty -  
  * Ubuntu:Vivid -  
  * Ubuntu:Xenial -  
  * Ubuntu:Yakkety -

**Comment by ovh-cds** [comment]
CDS Report build-deb#96.0 ✘  
* Stage 1
  * Debian:Jessie ✘  
  * Debian:Stretch ✘  
  * Debian:Wheezy ✘

**Comment by ovh-cds** [comment]
CDS Report build-deb#97.0 ✘  
* Stage 1
  * Debian:Jessie ✘  
  * Debian:Stretch ✘  
  * Debian:Wheezy ✘

**Pull request merged**